plugins/manipulator.cpp

```diff
@@ -542,94 +782,782 @@ class StringFormatter {
+                    }else{
+                        cur->topskilled+=
+                            static_cast<int>(wskill*sqrt(static_cast<double>(mskill)/static_cast<double>(5+avg_skill)));
+                    }
+                }
+
+                //martial
+                if( (group>12 && group < 16)
+                  ||col_skill==df::job_skill::MILITARY_TACTICS
+                  ||col_skill==job_skill::PACIFY){
+                  if(col_skill==df::job_skill::TEACHING){
+                      cur->martial+=(cur->martial*wskill)/(70*70*20);
+                  }else if(col_skill==df::job_skill::KNOWLEDGE_ACQUISITION){
+                      cur->martial+=wskill/3;
+                  }else{
+                      cur->martial+=wskill;
+                 }
+
+                  if(group<15 ||col_skill==df::job_skill::DISCIPLINE
+                    ||col_skill==df::job_skill::DODGING)
+                      cur->martial+=wskill/2; //extra important
+                }
+
+                //scholar
+                if(col_skill==df::job_skill::TEACHING
+                  ||col_skill==df::job_skill::CONCENTRATION
+                  ||col_skill==df::job_skill::WRITING
+                  ||col_skill==df::job_skill::READING
+                  ||col_skill==df::job_skill::KNOWLEDGE_ACQUISITION
+                  ||col_skill==df::job_skill::CONVERSATION
+                  ||group==21){
+                    cur->scholar+=wskill;
+                }
+
+                //performer
+                if( group == 19 && col_skill != df::job_skill::WRITING
+                ||col_skill==job_skill::COMEDY
+                ||col_skill==job_skill::FLATTERY
+                ||col_skill==job_skill::CONSOLE
+                ||col_skill==job_skill::PACIFY){
+                    if(group == 19)
+                        cur->performer+=wskill;
+                    else
+                        cur->performer+=wskill/2;
+                }
+            }
+        }
+        work_aptitude_avg = uinfo_avg_work_aptitude ;
+        skill_aptitude_avg = uinfo_avg_skill_aptitude;
+
+    }//end of scoring.. next comes notice string gen
+
+
+
+    //from cursecheck.cpp
+    string determineCurse(df::unit * unit)
+    {
+        string cursetype = "";
+        if(unit->curse_year == -1) return cursetype;
+
+        cursetype = "Shade "; //an unresting soul ghost was this
+
+        if(unit->flags3.bits.ghostly)
+            cursetype = "Ghost "; //yet not this
+
+        if( (unit->curse.add_tags1.bits.OPPOSED_TO_LIFE || unit->curse.add_tags1.bits.NOT_LIVING)
+            && !unit->curse.add_tags1.bits.BLOODSUCKER )
+            cursetype = "Zmbie ";
+
+        if(!unit->curse.add_tags1.bits.NOT_LIVING
+            && unit->curse.add_tags1.bits.NO_EAT
+            && unit->curse.add_tags1.bits.NO_DRINK
+            && unit->curse.add_tags2.bits.NO_AGING
+            )
+            cursetype = "Ncrmncr ";
```

Comment:
Do these need to be abbreviated?

Reply:
Quite, if we prefer a few of them to be able to fit into the available detail space.